src/migrate.rs

```diff
@@ -0,0 +1,139 @@
+// Config.json migration module
+//
+// Provides methods for migrating config.json fields based on remote directives
+// from /os/vX/config. Limits migrated fields based on os-config.json schema
+// whitelist.
+
+use crate::config_json::ConfigMap;
+use crate::remote::ConfigMigrationInstructions;
+use crate::schema::OsConfigSchema;
+use anyhow::Result;
+
+pub fn generate_config_json_migration(
+    schema: &OsConfigSchema,
+    migration_config: &ConfigMigrationInstructions,
+    config_json: &ConfigMap,
+) -> Result<ConfigMap> {
+    info!("Checking for config.json migrations...");
+
+    let mut new_config = config_json
+        .iter()
+        .map(|(k, v)| (k.clone(), v.clone()))
+        .collect::<ConfigMap>();
+
+    let keys_to_update = migration_config.update.keys().cloned().collect::<Vec<_>>();
+
+    for key in keys_to_update.iter().chain(migration_config.delete.iter()) {
+        if !schema.config.whitelist.contains(key) {
+            info!("Key `{}` not in whitelist, skipping", key);
+            continue;
+        }
+
+        // Key is marked for deletion
+        if migration_config.delete.contains(key) {
+            if config_json.contains_key(key) {
+                info!("Key `{}` found, will delete", key);
+                new_config.remove(key);
+            } else {
+                info!("Key `{}` not found for deletion, skipping", key);
```

Comment:
Maybe a `debug!` here instead?

Reply:
This is only for the bottom `info!` that skips, not the above one that actually deletes.